gpu/octree/src/cuda/radius_search.cu

```diff
@@ -229,33 +219,29 @@ namespace pcl
                     mask &= ~(1 << active_lane);              
 
                     //broadcast active_found_count                                
-                    if (active_lane == laneId)                
-                        storage.per_warp_buffer[warpId] = found_count;                                            
-                    int active_found_count = storage.per_warp_buffer[warpId];
+                    const int active_found_count = __shfl_sync(0xFFFFFFFF, found_count, active_lane);
 
-                    int node_idx = leaf & ~KernelPolicy::CHECK_FLAG;
+                    const int node_idx = leaf & ~KernelPolicy::CHECK_FLAG;
 
-                    //broadcast beg
+                    //broadcast beg and end
+                    int fbeg, fend;
                     if (active_lane == laneId)
-                        storage.per_warp_buffer[warpId] = batch.octree.begs[node_idx];                    
-                    int beg = storage.per_warp_buffer[warpId];
-
-                    //broadcast end
-                    if (active_lane == laneId)
-                        storage.per_warp_buffer[warpId] = batch.octree.ends[node_idx];
-                    int end = storage.per_warp_buffer[warpId];
+                    {
+                      fbeg = batch.octree.begs[node_idx];
+                      fend = batch.octree.ends[node_idx];
+                    }
+                    const int beg = __shfl_sync(0xFFFFFFFF, fbeg, active_lane);
+                    const int end = __shfl_sync(0xFFFFFFFF, fend, active_lane);
```

Comment:
does this work? if yes, I'm I incurring in any penalty for doing so?
```suggestion
                    const int beg = __shfl_sync(0xFFFFFFFF, batch.octree.begs[node_idx], active_lane);
                    const int end = __shfl_sync(0xFFFFFFFF, batch.octree.ends[node_idx], active_lane);
```

Reply:
This way it failed to run, with a bad memory reference or something like that.
Thats why I extracted it to a local variable, before sharing.

Reply:
Yeah, it gives an illegal memory access error. I think it's because `Batch` is not in the warp shared memory, it's coming from outside.